src/lib/serviceRoutines/postQueryContext.cpp

```diff
@@ -220,47 +159,282 @@ std::string postQueryContext
+    parseData.qcrs.res.release();
+    return;
   }
 
 
   //
-  // 6. Render QueryContextResponse from the providing application
+  // 5. Render QueryContextResponse from the providing application
   //
   char portV[16];
   snprintf(portV, sizeof(portV), "%d", port);
 
-  // Fill in the response from the redirection into the response to the originator of this request
-  QueryContextResponse* qcrsP = &parseDataP->qcrs.res;
+  // Fill in the response from the redirection into the response
+  qcrsP->fill(&parseData.qcrs.res);
+
 
   //
-  // Returning 'redirected to' in StatusCode::details
+  // 6. 'Fixing' StatusCode
   //
-  if ((qcrsP->errorCode.code == SccOk) || (qcrsP->errorCode.details == ""))
+  if (qcrsP->errorCode.code == SccNone)
   {
-    qcrsP->errorCode.details = std::string("Redirected to context provider ") + ip + ":" + portV + prefix;
+    qcrsP->errorCode.fill(SccOk);
   }
-  else if (qcrsP->errorCode.code == SccNone)
+
+  if ((qcrsP->contextElementResponseVector.size() == 1) && (qcrsP->contextElementResponseVector[0]->statusCode.code == SccContextElementNotFound))
   {
-    qcrsP->errorCode.fill(SccOk, std::string("Redirected to context provider ") + ip + ":" + portV + prefix);
+    qcrsP->errorCode.fill(SccContextElementNotFound);
   }
 
-  if (qcrsP->contextElementResponseVector.size() > 0)
+  
+  //
+  // 7. Freeing memory
+  //
+  parseData.qcr.res.release();
+  parseData.qcrs.res.release();
+}
+
+
+
+/* ****************************************************************************
+*
+* postQueryContext -
+*/
+std::string postQueryContext
+(
+  ConnectionInfo*            ciP,
+  int                        components,
+  std::vector<std::string>&  compV,
+  ParseData*                 parseDataP
+)
+{
+  //
+  // Convops calling this routine may need the response in digital
+  // So, the digital response is passed back in parseDataP->qcrs.res.
+  //
+  QueryContextResponse*       qcrsP = &parseDataP->qcrs.res;
+  QueryContextRequest*        qcrP  = &parseDataP->qcr.res;
+  std::string                 answer;
+  QueryContextRequestVector   requestV;
+  QueryContextResponseVector  responseV;
+
+
+  //
+  // 01. Call mongoBackend/mongoQueryContext
+  //
+  qcrsP->errorCode.fill(SccOk);
+  ciP->httpStatusCode = mongoQueryContext(qcrP, qcrsP, ciP->tenant, ciP->servicePathV, ciP->uriParam);
+
+
+  //
+  // 02. Normal case (no requests to be forwarded)
+  //
+  // If the result from mongoBackend is a 'simple' result without any forwarding needed (this is the
+  // normal case), then there is no need to execute the special treatment that forwarding needs.
+  // It's better not to mix the two very different treatments of the output from mongoBackend.
+  //
+  // Now, the request is 'simple' if all providingApplicationLists of the ContextElements are empty and
+  // no ContextAttribute has any providingApplication.
+  //
+  bool requestDone = true;
+  for (unsigned int ix = 0 ; ix < qcrsP->contextElementResponseVector.size(); ++ix)
```

Comment:
Maybe the "noneCpr" block (from 252 to 271) could be moved to a function for clarity (the `break` statement will become `return`).